src/lib.rs

```diff
@@ -95,9 +103,91 @@ pub mod g_rpc {
     }
 }
 
+// A collection of functions for client side connections to the proxy server
 impl NetworkConfig {
     pub fn load(path: &str) -> Result<NetworkConfig, Box<dyn Error>> {
         let file = std::io::BufReader::new(File::open(path)?);
         Ok(serde_json::from_reader(file)?)
     }
+
+    /// get_client is an internal function to obtain the uds endpoint
+    ///
+    /// # Arguments
+    ///
+    /// * `p`: path to uds
+    ///
+    /// returns: Result<NetavarkProxyClient<Channel>, Status>
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///
+    /// ```
+    async fn get_client(p: String) -> Result<NetavarkProxyClient<Channel>, Status> {
+        let endpoint = match Endpoint::try_from("http://[::1]:10000") {
+            Ok(ep) => ep,
+            Err(e) => return Err(Status::internal(e.to_string())),
+        };
+
+        let channel = match endpoint
+            .connect_with_connector(service_fn(move |_: Uri| {
+                let pp = p.clone();
+                debug!("using uds path: {}", pp);
+                UnixStream::connect(pp)
+            }))
+            .await
+        {
+            Ok(c) => c,
+            Err(e) => return Err(Status::internal(e.to_string())),
+        };
```

Comment:
```suggestion
            .await
            .map_err(|e| Status::internal(e.to_string()))?;
```

Reply:
(There's various other places you're doing an explicit `match` - nothing wrong with that at all, but I think combinators in most of these cases are more elegant and concise)